juju/model.py

```diff
@@ -1536,14 +1536,25 @@ def set_constraints(self, constraints):
         """
         raise NotImplementedError()
 
-    def get_action_output(self, action_uuid, wait=-1):
+    async def get_action_output(self, action_uuid, wait=-1):
         """Get the results of an action by ID.
 
         :param str action_uuid: Id of the action
         :param int wait: Time in seconds to wait for action to complete
-
+        :return dict: Output from action
+        :raises: :class:`JujuError` if invalid action_uuid
         """
-        raise NotImplementedError()
+        action_facade = client.ActionFacade.from_connection(
+            self.connection()
+        )
+        entity = [{'tag': tag.action(action_uuid)}]
+        action_output = await action_facade.Actions(entity)
+        # ActionResult.output is None if the action produced no output
+        if action_output.results[0].output:
```

Comment:
If it is likely to be None, then the check should be `is None` and not just false-y.